tests/modules/blockonkeys.c

```diff
@@ -145,6 +147,9 @@ int bpop_reply_callback(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
 
     RedisModule_Assert(fsl->length);
     RedisModule_ReplyWithLongLong(ctx, fsl->list[--fsl->length]);
+
+    /* I'm lazy so i'll replicate a potentially blocking command, it shouldn't block in this flow. */
+    RedisModule_ReplicateVerbatim(ctx);
```

Comment:
Maybe we should assert then? On `fsl_bpop` if we decide to block then we are not a replica?

Reply:
maybe even go further and assert somewhere in blocked.c that a replica never blocks its master client?